npm/puppeteer/README.md

```diff
@@ -0,0 +1,360 @@
+# @cypress/puppeteer
+
+Utilize Puppeteer's browser API within Cypress with a single command.
+
+# Table of Contents
+
+- [Installation](#installation)
+- [Compatibility](#compatibility)
+- [Usage](#usage)
+- [API](#api)
+- [Examples](#examples)
+- [Contributing](#contributing)
+- [Changelog](./CHANGELOG.md)
+
+# Installation
+
+## npm
+
+```sh
+npm install --save-dev @cypress/puppeteer
+```
+
+## yarn
+
+```sh
+yarn add @cypress/puppeteer
+```
+
+## With TypeScript
+
+Add the following in `tsconfig.json`:
+
+```json
+{
+  "compilerOptions": {
+    "types": ["cypress", "@cypress/puppeteer"]
+  }
+}
+```
+
+## Compatibility
+
+`@cypress/puppeteer` requires Cypress version 13.6.0 or greater.
+
+Only Chromium-based browsers (e.g. Chrome, Chromium, Electron) are supported.
+
+## Usage
+
+`@cypress/puppeteer` is set up in your Cypress config and support file, then executed in your spec. See [API](#api) and [Examples](#examples) below for more details.
+
+While the `cy.puppeteer()` command is executed in the browser, the majority of the Puppeteer execution is run in the Node process via your Cypress config. You pass a string message name to `cy.puppeteer()` that indicates which message handler to execute in the Cypress config. This is similar to how [cy.task()](on.cypress.io/task) operates.
+
+In your Cypress config (e.g. `cypress.config.ts`):
+
+```typescript
+import { setup } from '@cypress/puppeteer/plugin'
+
+export default defineConfig({
+  e2e: {
+    setupNodeEvents (on) {
+      setup({
+        on,
+        onMessage: {
+          async switchToTabAndGetContent (browser) {
+            // ...
```

Comment:
@chrisbreiding I would probably put an explicit comment here that says you need to now write Puppeteer code to execute what you want. I think someone might copy/paste this and assume it should do what it says. 🙂

Reply:
Good idea. Addressed in https://github.com/cypress-io/cypress/pull/28370/commits/1d8356a92cf050de9c63813c72115bfcb160418e

Reply:
Was this addressed? I don't see an explicit comment?

Reply:
Ah good catch, I had put the comment [here](https://github.com/cypress-io/cypress/commit/1d8356a92cf050de9c63813c72115bfcb160418e#diff-bad8df9ac3ed2d72ad00ba7b71b5a9a6f51364a91eaf05c7ba1c9b1895fab279R225) but it's a good idea to add it in this spot as well.

Reply:
Actually addressed this time [here](https://github.com/cypress-io/cypress/pull/28370/commits/beb85942ff50c82e10405d142cc6b0bc820580c6#diff-bad8df9ac3ed2d72ad00ba7b71b5a9a6f51364a91eaf05c7ba1c9b1895fab279R65)